openedx/features/qverse_features/registration/signals.py

```diff
@@ -134,58 +196,139 @@ def _write_status_on_csv_file(filename, output_file_rows):
+        """
+        Validates that given row contains all the required fields.
 
-    if len(row['mobile']) > MOBILE_NUMBER_MAX_LENGTH:
-        errors.append('Mobile number is more than {} characters long.'.format(MOBILE_NUMBER_MAX_LENGTH))
+        Arguments:
+            row (dict): A dictionary containing user information
 
-    try:
-        if int(row['levelid']) < 1 or int(row['levelid']) > MAX_LEVEL_CHOICES:
-            errors.append('Level ID must be greater than 0 and smaller than {}.'.format(MAX_LEVEL_CHOICES+1))
-    except ValueError:
-        errors.append('Level ID is not an integer value.')
+        Returns:
+            error (str): Contains error related to provided row
+        """
+        OPTIONAL_FIELDS = [
+            'mobile', 'othername', 'status', 'error', 'password'
+        ]
+        required_values = [value.strip() if value else value for (key, value) in row.items() if key not in OPTIONAL_FIELDS]
+        all_values_available = all(required_values)
 
-    try:
-        if int(row['programmeid']) < 1 or int(row['programmeid']) > MAX_PROGRAMME_CHOICES:
-            errors.append('Programme ID must be greater than 0 and smaller than {}.'.format(MAX_PROGRAMME_CHOICES+1))
-    except ValueError:
-        errors.append('Programme ID is not an integer value.')
-
-    if errors:
-        # to write multiple lines
-        # in a single cell of csv
-        errors = '"{}"'.format('\n'.join(errors))
-        return False, errors
-
-    return True, None
+        if not all_values_available:
+            return 'Please provide values for all required fields.'
+
+    def _validate_field_values(self):
+        """
+        Validates the values of all the given fields of single CSV row.
+        """
+        self._validate_registration_number()
+        self._validate_email_address()
+        self._validate_unique_email_constraint()
+        self._validate_first_name()
+        self._validate_surname()
+        self._validate_other_name()
+        self._validate_current_level()
+        self._validate_department_number()
+        self._validate_programme_id()
+        self._validate_mobile_number()
+
+    def _validate_email_address(self):
+        try:
+            validate_email = EmailValidator()
+            validate_email(self.email)
+        except ValidationError:
+            self.errors.append('Please provide a valid email address.')
+
+    def _validate_unique_email_constraint(self):
+        try:
+            user = User.objects.get(email=self.email)
+            if user.username != self.regno:
+                self.errors.append('{} is already associated with another user account.'.format(self.email))
+        except User.DoesNotExist:
+            pass
+
+    def _validate_department_number(self):
+        try:
+            if not Department.objects.filter(number=self.department_id).exists():
+                self.errors.append('Department with this {} number does not exist.'.format(self.department_id))
```

Comment:
```suggestion
                self.errors.append('Department number {} does not exist.'.format(self.department_id))
```
???
Or use anything you prefer. I just find that "this <number> number" (say "this 10 number") odd.